extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSources.java

```diff
@@ -140,6 +143,38 @@ public AgroalDataSource doCreateDataSource(String dataSourceName) {
         // we first make sure that all available JDBC drivers are loaded in the current TCCL
         loadDriversInTCCL();
 
+        if (dataSourceJdbcBuildTimeConfig.tracing) {
+            dataSourceJdbcRuntimeConfig.tracing.enabled = Optional
```

Comment:
Use a local variable, don't override the config value. You never know what could happen.